nautobot/extras/models/groups.py

```diff
@@ -853,6 +864,53 @@ def get_group_members_url(self):
         """Return the group members URL."""
         return self.group.get_group_members_url()
 
+    def get_siblings(self, include_self=False):
+        """Return group memberships that share the same parent group."""
+        siblings = DynamicGroupMembership.objects.filter(parent_group=self.parent_group)
+        if include_self:
+            return siblings
+        return siblings.exclude(pk=self.pk)
+
+    def _get_next_or_previous_by_weight(self, is_next):
+        """Get siblings and return the next/previous based on `is_next`."""
+        siblings = self.get_siblings(include_self=True)
+        if not is_next:
+            siblings = reversed(siblings)
+
+        siblings = list(siblings)
+        self_idx = siblings.index(self)
+
+        try:
+            return siblings[self_idx + 1]
+        except IndexError:
+            raise self.DoesNotExist(f"{self.__class__._meta.object_name} matching query does not exist.")
+
+    def get_next_by_weight(self):
```

Comment:
nit: i'd find `get_[next|previous]_sibling` to be a clearer name, `by_weight` is implicit IMO